README.md

```diff
@@ -1578,6 +1579,43 @@ The [Eio_unix][] module provides features for using Eio with OCaml's Unix module
 In particular, `Eio_unix.run_in_systhread` can be used to run a blocking operation in a separate systhread,
 allowing it to be used within Eio without blocking the whole domain.
 
+### Domainslib
+
+For certain compute-intensive tasks it may be useful to send work to a pool of [Domainslib][] worker domains.
+You can resolve an Eio promise from non-Eio domains (or systhreads), which provides an easy way to retrieve the result.
+For example:
+
+<!-- $MDX skip -->
+```ocaml
+open Eio.Std
+
+let pool = Domainslib.Task.setup_pool ~num_domains:2 ()
+
+let fib n = ... (* Some Domainslib function *)
+
+let run_in_pool fn x =
+  let result, set_result = Promise.create () in
+  let _ : unit Domainslib.Task.promise = Domainslib.Task.async pool (fun () ->
```

Comment:
I initially assumed that the `send` that happens internally with `Task.async` would be safe to call from non-domainslib domains, but as I'm now looking into adding POC domain local await support to domainslib, I realize that this works somewhat accidentally.

When you create the pool

```ocaml
let pool = Domainslib.Task.setup_pool ~num_domains:2 ()
```

the requsted `n=2` domains are started.  Additionally an internal "multi channel" is created with `n+1` capacity.  The `+1` reserves an extra internal "channel" for the domain that calls `Task.run`.

When `Task.async` is called, it looks up the internal channel for the current domain.  If no channel has already been allocated for the current domain, it then allocates a channel for the domain from the reserved `n+1`.  This works as long as `Task.async` is only called from a single additional domain.  If you had two (or more) Eio domains, then only one of those could to call `Task.async`.

Reply:
Interesting. But doesn't `Task.run` have the same limitation, then?

Reply:
Yes, I think it has the same limitation.

At any rate, I'm looking into how to potententially change that to be more domain safe.